src/fsharp/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.Utilities.fs

```diff
@@ -129,32 +92,56 @@ module internal Utilities =
         // 3. We are loaded in-process to some other application ... Eg. try .net
         //    See if the host is dotnet.exe ... from net5.0 on this is fairly unlikely
         // 4. If it's none of the above we are going to have to rely on the path containing the way to find dotnet.exe
-        //
+        let fileExists pathToFile =
+            try
+                if File.Exists(pathToFile) then 
+                    true
+                else
+                    false
+            with | _ -> false
+        // Use the path to search for dotnet.exe
+        let probePathForDotnetHost () =
+            let paths =
+                let p = Environment.GetEnvironmentVariable("PATH")
+                if not(isNull p) then p.Split(';') 
+                else [||]
+            paths |> Array.tryFind (fun f -> fileExists (Path.Combine(f, dotnet)))
+
         if isRunningOnCoreClr then
             match (Environment.GetEnvironmentVariable("DOTNET_HOST_PATH")) with
-            | value when not (String.IsNullOrEmpty(value)) ->
-                Some value                           // Value set externally
+            // Value set externally
+            | value when not (String.IsNullOrEmpty(value)) && fileExists value -> Some value
             | _ ->
                 // Probe for netsdk install, dotnet. and dotnet.exe is a constant offset from the location of System.Int32
-                let dotnetLocation =
-                    let dotnetApp =
-                        let platform = Environment.OSVersion.Platform
-                        if platform = PlatformID.Unix then "dotnet" else "dotnet.exe"
+                let candidate =
                     let assemblyLocation = Path.GetDirectoryName(typeof<Int32>.GetTypeInfo().Assembly.Location)
-                    Path.GetFullPath(Path.Combine(assemblyLocation, "../../..", dotnetApp))
-
-                if File.Exists(dotnetLocation) then
-                    Some dotnetLocation
+                    Path.GetFullPath(Path.Combine(assemblyLocation, "../../..", dotnet))
```

Comment:
You can pass multiple args to `Combine()`, then you don't have to worry about the separator.

```suggestion
                    Path.GetFullPath(Path.Combine(assemblyLocation, "..", "..", "..", dotnet))
```